### ISSUE TITLE
Creeps and resources vanishing

### ISSUE THREAD

**Issue by Whoops** [screeps/screeps#42]
Sometimes multiple times a day, sometimes every few days, all the Creeps and energy in my private server will simply vanish. I'm not 100% sure they are connected, but everytime after I notice all the creeps are gone, there is this entry in the `backend.log`

```
Running cronjob 'roomsForceUpdate'
Unhandled rejection: Error: Can't set headers after they are sent.
    at ServerResponse.setHeader (_http_outgoing.js:359:11)
    at ServerResponse.header (/usr/local/lib/node_modules/screeps/node_modules/express/lib/response.js:719:10)
    at ServerResponse.send (/usr/local/lib/node_modules/screeps/node_modules/express/lib/response.js:164:12)
    at ServerResponse.json (/usr/local/lib/node_modules/screeps/node_modules/express/lib/response.js:250:15)
    at /usr/local/lib/node_modules/screeps/node_modules/q-json-response/q-json-response.js:17:14
    at _rejected (/usr/local/lib/node_modules/screeps/node_modules/q/q.js:844:24)
    at /usr/local/lib/node_modules/screeps/node_modules/q/q.js:870:30
    at Promise.when (/usr/local/lib/node_modules/screeps/node_modules/q/q.js:1122:31)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/screeps/node_modules/q/q.js:788:41)
    at /usr/local/lib/node_modules/screeps/node_modules/q/q.js:604:44
```

as far as I can tell there is nothing else interesting in the logs. Running screeps 2.2.1 on Node 7.3.0 on FreeBSD.

**Comment by DeathTech154** [comment]
I get that error too.
I think the cpu clogging slowly breaking the colonies is more to be blamed though. I only encounter empty colony syndrome when the cpu spirals our ridiculously high. My 5 - 9 cpu critters were doing 90 cpu / tick for no reason that I can find. This killed the colony.

**Comment by iKlem** [comment]
For me i don't think the error is linked to the vanishing. This is an HTTP error when we try to send headers when they are already sent. [](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers)

**Comment by Whoops** [comment]
I actually discovered what was going on last night. I happened to glance over when there was a red creep attacking my creeps, which was weird, because the bots on my server are pretty brain-dead. I clicked it, and it said something along the lines of "This creep is unhappy with your harvesting activity in this area". He made pretty short work of all my creeps and then vanished.

Googling around I could find zero mention of this related to screeps though, so I have no idea if it happens on the main game or if this "feature" is just private servers. It's certainly completely undocumented as far as I can tell, and since notifications don't work with private servers it's pure luck I ever saw it and figured it out.
